crates/router/src/core/user/theme.rs

```diff
@@ -0,0 +1,225 @@
+    );
+
+    let db_theme = state
+        .global_store
+        .insert_theme(new_theme)
+        .await
+        .to_duplicate_response(UserErrors::ThemeAlreadyExists)?;
+
+    theme_utils::upload_file_to_theme_bucket(
+        &state,
+        &theme_utils::get_theme_file_key(&db_theme.theme_id),
+        request
+            .theme_data
+            .encode_to_vec()
+            .change_context(UserErrors::InternalServerError)?,
+    )
+    .await?;
+
+    let file = theme_utils::retrieve_file_from_theme_bucket(
+        &state,
+        &theme_utils::get_theme_file_key(&db_theme.theme_id),
+    )
+    .await?;
+
+    let parsed_data = file
+        .to_bytes()
+        .parse_struct("ThemeData")
+        .change_context(UserErrors::InternalServerError)?;
+
+    Ok(ApplicationResponse::Json(theme_api::GetThemeResponse {
+        theme_id: db_theme.theme_id,
+        entity_type: db_theme.entity_type,
+        tenant_id: db_theme.tenant_id,
+        org_id: db_theme.org_id,
+        merchant_id: db_theme.merchant_id,
+        profile_id: db_theme.profile_id,
+        theme_name: db_theme.theme_name,
+        theme_data: parsed_data,
+    }))
+}
+
+pub async fn update_theme(
+    state: SessionState,
+    theme_id: String,
+    request: theme_api::UpdateThemeRequest,
+) -> UserResponse<theme_api::GetThemeResponse> {
+    let db_theme = state
+        .global_store
+        .find_theme_by_lineage(request.lineage)
+        .await
+        .to_not_found_response(UserErrors::ThemeNotFound)?;
+
+    if theme_id != db_theme.theme_id {
+        return Err(UserErrors::ThemeNotFound.into());
+    }
+
+    theme_utils::upload_file_to_theme_bucket(
+        &state,
+        &theme_utils::get_theme_file_key(&db_theme.theme_id),
+        request
+            .theme_data
+            .encode_to_vec()
+            .change_context(UserErrors::InternalServerError)?,
+    )
+    .await?;
+
+    let file = theme_utils::retrieve_file_from_theme_bucket(
+        &state,
+        &theme_utils::get_theme_file_key(&db_theme.theme_id),
+    )
+    .await?;
+
+    let parsed_data = file
+        .to_bytes()
+        .parse_struct("ThemeData")
+        .change_context(UserErrors::InternalServerError)?;
+
+    Ok(ApplicationResponse::Json(theme_api::GetThemeResponse {
+        theme_id: db_theme.theme_id,
+        entity_type: db_theme.entity_type,
+        tenant_id: db_theme.tenant_id,
+        org_id: db_theme.org_id,
+        merchant_id: db_theme.merchant_id,
+        profile_id: db_theme.profile_id,
+        theme_name: db_theme.theme_name,
+        theme_data: parsed_data,
+    }))
+}
+
+pub async fn delete_theme(
+    state: SessionState,
+    theme_id: String,
+    lineage: ThemeLineage,
+) -> UserResponse<()> {
+    state
+        .global_store
+        .delete_theme_by_lineage_and_theme_id(theme_id.clone(), lineage)
+        .await
+        .to_not_found_response(UserErrors::ThemeNotFound)?;
+
+    // TODO (#6717): Delete theme folder from the theme storage.
```

Comment:
if we cannot do delete, can we rename the directory in s3 and prefix/suffix it with some deleted identifier?

Reply:
Yeah, will check this.